### ISSUE TITLE
Gradient normalisation used in `biohub_optimizer` doesn't match ESMFold2 methodology

### ISSUE THREAD

**Issue by magnusross** [escalante-bio/mosaic#84]
The binder sequence optimisation method proposed in the [ESMFold2 paper](https://www.biorxiv.org/content/10.64898/2026.06.03.729735v1) (Algorithm 11) normalizes the structural and sequence parts of the gradient separately in logit space (i.e. after pulling each gradient back from probability space to logit space):

```
g_ref = N(Jᵀ g_struct) + 0.15 · N(Jᵀ g_seq)
```

where $J=(\text{diag}(p)-pp^\top)/T$ is the softmax jacobian, and N is the a gradient normalizer.

The mosaic implemenation of the method (`biohub_optimiser`) instead normalizes each branch in probability space, combines the branches, applies the softmax pullback once, and then normalizes the combined result:

```
g_mosaic = N(Jᵀ[U(g_struct) + 0.15 · U(g_seq)])
```

where \(U\) denotes probability-space normalization.

Normalization and the softmax pullback do not commute. Consequently, Mosaic does not preserve the intended fixed 0.15 relative weighting after the pullback. If

```
a = JᵀU(g_struct)
b = JᵀU(g_seq)
```

then Mosaic’s effective relative weight is:

```
w_eff = 0.15 · ‖b‖ / ‖a‖
```

rather than a fixed 0.15. This ratio changes as the probability distribution sharpens during temperature annealing, causing Mosaic and the reference implementation to follow different optimization trajectories.

**Comment by nboyd** [comment]
Great catch, thanks @magnusross ! I added a dedicated optimizer/loss for this case (demo in the minibinder notebook) -- if you can come up with a more elegant way to do this kind of thing in general let me know
